### DISCUSSION TITLE
Delta ASDA driver,  switch modes from input capability?

### DISCUSSION THREAD

**Discussion by DannoDempsey** [linuxcnc-ethercat/linuxcnc-ethercat#449] [Ideas]
Hi,

Similar to RodW, I am working on the custom homecomp component to realize the use of the servo's internal homing routine. Is it possible to modify the Delta ASDA driver to the ability to switch modes from an external input. i.e. a bit that can be set true to switch from CSP mode => HOMING mode or CSV mode => HOME mode?  I've accomplished this by breaking everything out in the xml file  and ladder logic in the past but this might be a worthy addition to the driver. Unfortunately, I know nothing about writing drivers so I am way out of my element here.   

Also, the delta servo's that I use have brakes which require a 500ms delay to turn on and off during mode switches. Thought?

**Comment by zmrdko** [comment]
Hello, did sort out this issue?

**Comment by DannoDempsey** [comment]
Hi,
Unfortunately, I've been a bit side tracked working on a custom gui for
ethercat configs (lathes) and added ethercat homing flag to my RIP version
of LinuxCNC so I will be circling back around in the next few weeks for
actual testing of the internal homing routines.
I just don't know enough about drivers to start messing around, so I've
always switched modes via classic ladder and custom m-codes.

I think this capability would be useful but I'm not sure how difficult it
would be to implement... and if so is it worth it?

Dan
